src/Encamina.Enmarcha.SemanticKernel.Abstractions/MemoryStoreHandlerBase.cs

```diff
@@ -51,9 +65,10 @@ public virtual async Task<string> GetCollectionNameAsync(string collectionId, Ca
             LastAccessUtc = DateTime.UtcNow,
         };
 
-        if (!await MemoryManager.MemoryStore.DoesCollectionExistAsync(collectionName, cancellationToken))
+        if (!await MemoryStoreExtender.MemoryStore.DoesCollectionExistAsync(collectionName, cancellationToken))
         {
-            await MemoryManager.MemoryStore.CreateCollectionAsync(collectionName, cancellationToken);
+            await MemoryStoreExtender.MemoryStore.CreateCollectionAsync(collectionName, cancellationToken);
+            MemoryStoreExtender.OnMemoryStore(new() { EventType = Events.MemoryStoreEventTypes.CreateCollection, CollectionName = collectionName });
         }
 
```

Comment:
Suppose that the collection is already created, thus lines 68 to 71 are not executed (which is totally OK), but I'm still missing an event that I could hook to when getting a collection's name.